src/iproject.ts

```diff
@@ -407,6 +407,46 @@ export class IProject {
     }
   }
 
+  public async moveLibraryList(library: string, type: LibraryType, direction: Direction) {
+    const unresolvedState = await this.getUnresolvedState();
+    let attribute: keyof IProjectT;
+
+    if (unresolvedState) {
+      if (type === LibraryType.preUserLibrary || LibraryType.postUserLibrary) {
+
+          attribute = type === LibraryType.preUserLibrary ? 'preUsrlibl' : 'postUsrlibl';
+
+          let libIndex = -1;
```

Comment:
Here I think we can just do what you did for the reordering of the include paths:
```typescript
const index = unresolvedState[attribute] ? unresolvedState[attribute].indexOf(library) : -1;
if (index > -1) {
    ...
```